src/Subscription.ts

```diff
@@ -13,31 +11,36 @@ export class Subscription {
   public isUnsubscribed: boolean = false;
 
   constructor(_unsubscribe?: () => void) {
-    if (_unsubscribe) {
+    // Check `_unsubscribe`to avoid override `this._unsubscribe` accidentally.
+    // Dn't call `super(_unsubscribe)` from a derived class.
+    if (!!_unsubscribe) {
```

Comment:
If you're asserting "truthiness" here, you can just do it, no need for the `!!`. If you truly need a boolean, then `Boolean(x)` is probably more efficient than two operators back to back (but I'm speculating)